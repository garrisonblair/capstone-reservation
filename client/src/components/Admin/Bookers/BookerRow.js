import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Table, Button, Icon } from 'semantic-ui-react';
import BookerModal from './BookerModal';


class BookerRow extends Component {
  state = {
    openModal: false,
  }

  onClickEditButton = () => {
    this.setState({ openModal: true });
  }

  closeModal = () => {
    const { syncBookers } = this.props;

    this.setState({ openModal: false });
    syncBookers();
  }

  render() {
    const { booker } = this.props;
    const { openModal } = this.state;
    return (
      <Table.Row key={booker.username}>
        <Table.Cell textAlign="center">
          {booker.username}
        </Table.Cell>
        <Table.Cell textAlign="center">
          {`${booker.first_name} ${booker.last_name}`}
        </Table.Cell>
        <Table.Cell textAlign="center">
          {booker.email}
        </Table.Cell>
        <Table.Cell textAlign="center">
          {booker.is_superuser ? <Icon name="check circle" color="green" /> : <Icon name="times circle" color="red" />}
        </Table.Cell>
        <Table.Cell textAlign="center">
          {booker.is_staff ? <Icon name="check circle" color="green" /> : <Icon name="times circle" color="red" />}
        </Table.Cell>
        <Table.Cell textAlign="center">
          {booker.is_active ? <Icon name="check circle" color="green" /> : <Icon name="times circle" color="red" />}
        </Table.Cell>
        <Table.Cell>
          <Button icon="edit" className="edit-button" onClick={this.onClickEditButton} />
        </Table.Cell>
        {openModal ? <BookerModal show booker={booker} onClose={this.closeModal} /> : null}
      </Table.Row>
    );
  }
}

BookerRow.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  booker: PropTypes.object.isRequired,
  // eslint-disable-next-line react/no-unused-prop-types
  syncBookers: PropTypes.func.isRequired,
};
export default BookerRow;
