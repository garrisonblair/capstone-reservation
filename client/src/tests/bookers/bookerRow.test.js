/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import BookerRow from '../../components/Admin/Bookers/BookerRow';

function setup() {
  const booker = {
    first_name: 'first',
    last_name: 'last',
    username: 'username',
    email: 'test@email.com',
    booker_profile: {
      privilege_categories: [],
    },
  };
  const syncBookers = () => { };

  return { booker, syncBookers };
}

it('Shows bookers table', () => {
  const { booker, syncBookers } = setup();
  const tree = renderer
    .create(
      <BookerRow
        booker={booker}
        syncBookers={syncBookers}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
