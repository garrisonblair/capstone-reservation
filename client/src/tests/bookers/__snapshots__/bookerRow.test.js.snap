// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows bookers table 1`] = `
<tr
  className=""
>
  <td
    className="center aligned"
  >
    username
  </td>
  <td
    className="center aligned"
  >
    first last
  </td>
  <td
    className=""
  >
    <button
      className="ui icon button edit-button"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="edit icon"
      />
    </button>
  </td>
</tr>
`;
